pyro/poutine/trace_poutine.py

```diff
@@ -41,16 +41,24 @@ class TraceMessenger(Messenger):
     We can also use this for visualization.
     """
 
-    def __init__(self, graph_type=None):
+    def __init__(self, graph_type=None, param_only=None):
         """
         :param string graph_type: string that specifies the type of graph
             to construct (currently only "flat" or "dense" supported)
         """
         super(TraceMessenger, self).__init__()
         if graph_type is None:
             graph_type = "flat"
+        if param_only is None:
+            param_only = False
         assert graph_type in ("flat", "dense")
         self.graph_type = graph_type
+        self.param_only = param_only
+        self.trace = Trace(graph_type=self.graph_type)
+
+    def __enter__(self):
```

Comment:
Does this require any new usage patterns, other than the one use in `poutine.trace` below?

Reply:
This is necessary to reset `self.trace` so that `poutine.trace` can be used as a context manager